url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+			LinkedList<Long> buyOrderID = new LinkedList<Long>();
+			LinkedList<Long> sellOrderID = new LinkedList<Long>();
+			LinkedList<Long> OrderID;
+			lecMSReverse strategy = new lecMSReverse();
+			strategy.setThreShold(Threshold);
+			strategy.setAmountToTrade(stockAmount); 
+
+			long currentID = 0;
+			while (set.next()){
+				tmp = set.getString(5);
+				tmpType = set.getString(14);
+				tmpTime = set.getTime(3);
+				if(tmp.equalsIgnoreCase("ENTER")){
+					tmpPrice = set.getDouble(6);
+					tmpVol = set.getInt(7);
+					if(tmpType.equalsIgnoreCase("B")){
+						tmpID = set.getLong(12);
+						myBidList.add(tmpID,tmpPrice,tmpVol,tmpTime);
+						bidFirstList.add(new GraphData(myBidList.get(0).getPrice(),tmpTime));
+					}else if(tmpType.equalsIgnoreCase("A")){
+						tmpID = set.getLong(13);
+						myAskList.add(tmpID,tmpPrice,tmpVol,tmpTime);
+						askFirstList.add(new GraphData(myAskList.get(0).getPrice(),tmpTime));
+					}
+
+					if(tmpTime.after(startTime) && tmpTime.before(endTime)){
+						tmpCount = matchTrade(myBidList,myAskList,completedTrade,tmpTime,strategyTrade);
+
+						for(int i = 1; i <= tmpCount;i++){
+							strategy.addTrade(completedTrade.get(completedTrade.size() - (tmpCount - i) - 1).getPrice());
+							signalList = strategy.generateSignalList(myBidList, myAskList);
+							strategyTrade = new LinkedList<ResultData>();
+							OrderID = new LinkedList<Long>();
+							while(!signalList.isEmpty()){
+								tmpSignal = signalList.poll();
+								currentID--;
+								if(tmpSignal.getType().equalsIgnoreCase("sell")){
+									count++;
+									myAskList.add(currentID,tmpSignal.getPrice(),tmpSignal.getQuantity(),tmpTime);
+									askFirstList.add(new GraphData(myAskList.get(0).getPrice(),tmpTime));
+									strategyAsk.add(new GraphData(tmpSignal.getPrice(),tmpTime));
+									matchTrade(myBidList,myAskList,completedTrade,tmpTime,strategyTrade);
+									sellOrderID.add(currentID);
+								}else if (tmpSignal.getType().equalsIgnoreCase("buy")){
+									count++;
+									myBidList.add(currentID,tmpSignal.getPrice(),tmpSignal.getQuantity(),tmpTime);
+									bidFirstList.add(new GraphData(myBidList.get(0).getPrice(),tmpTime));
+									strategyBid.add(new GraphData(tmpSignal.getPrice(),tmpTime));
+									matchTrade(myBidList,myAskList,completedTrade,tmpTime,strategyTrade);
+									buyOrderID.add(currentID);
+								}
+								OrderID.add(currentID);
+							}
+							strategyCount += strategyTrade.size();
+							strategy.getSTrade(strategyTrade);
+							strategy.getReceiptList(OrderID);
+						}
+					}
+				}else if (tmp.equalsIgnoreCase("AMEND")){
+					updateLines++;
+					tmpPrice = set.getInt(6);
+					tmpVol = set.getInt(7);
+					if(tmpType.equalsIgnoreCase("B")){
+						tmpID = set.getLong(12);
+						myBidList.update(tmpID,tmpPrice,tmpVol,tmpTime);
+					}else if(tmpType.equalsIgnoreCase("A")){
+						tmpID = set.getLong(13);
+						myAskList.update(tmpID,tmpPrice,tmpVol,tmpTime);
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 405:411, 560:566.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.